scipy/sparse/csgraph/_laplacian.py

```diff
@@ -43,9 +48,28 @@ def laplacian(csgraph, normed=False, return_diag=False, use_out_degree=False):
     Notes
     -----
     The Laplacian matrix of a graph is sometimes referred to as the
-    "Kirchoff matrix" or the "admittance matrix", and is useful in many
+    "Kirchoff matrix" or just the "Laplacian", and is useful in many
     parts of spectral graph theory. In particular, the eigen-decomposition
-    of the laplacian matrix can give insight into many properties of the graph.
+    of the Laplacian can give insight into many properties of the graph, e.g.,
+    is commonly used for spectal data enmedding and clustering.
+
+    Unless the `inplace=True` parameter is used, the constructed Laplacian
+    doubles the memory use.
+
+    The Laplacian preserved the `dtype` and the format of the input
+    adjacency matrix, except that sparse 'lil' and 'dok' formats are
+    converted into `coo` thus doubling the memory even with `inplace=True`
```

Comment:
```suggestion
    converted into `coo` thus doubling the memory even with ``inplace=True``
```
I am not sure about the rendering the matrix types. It should at least be consistent and if possible link.

Reply:
This is what the original code actually does. I have not changed it, only documented.
 
I have no clue why  sparse 'lil' and 'dok' formats are converted into `coo` but changing it should be a different PR also dealing with not being backward compatible.

Reply:
My current description of the sparse format conversion logic is inaccurate. It appears from the code that all sparse formats are converted to coo, except for dia. The code logic is strange, so I am unsure if this is intentional or a bug.

Reply:
So what did you do in the end?

Reply:
I just updated the docstring to reflect my understanding of what the code actually does with various sparse formats, without making any related changes in the code, so it's fully backward compatible.